tests/helper/helper_oc.go

```diff
@@ -481,3 +481,11 @@ func (oc *OcRunner) GetServices(namespace string) string {
 	output := string(session.Wait().Out.Contents())
 	return output
 }
+
+// VerifyResourceDeleted verifies if the given resource is deleted from cluster
+func (oc *OcRunner) VerifyResourceDeleted(resourceType, resourceName, namespace string) {
+	session := CmdRunner(oc.path, "get", resourceType, "--namespace", namespace)
+	Eventually(session).Should(gexec.Exit(0))
+	output := string(session.Wait().Out.Contents())
```

Comment:
> I think this should be Err.Contents() because of kubernetes/kubectl#847
I am checking for contents, so i guess its ok.
Expect(output).NotTo(ContainSubstring(resourceName))

@adisky Perticularly in this case you should not use ```Out.Contents()``` because of the open issue kubernetes/kubectl#847 as mentioned by @mik-dass 

Actually you are verifying ```""``` with ```not to contains resourceName```. So the result comes always true because ```""``` never contains anything let say the resource name.  
```
func (oc *OcRunner) VerifyResourceDeleted(resourceType, resourceName, namespace string) {
	session := CmdRunner(oc.path, "get", resourceType, "--namespace", namespace)
	Eventually(session).Should(gexec.Exit(0))
	output := string(session.Wait().Err.Contents())
	fmt.Println("StdOut is :", output)
	Expect(output).NotTo(ContainSubstring(resourceName))
}

Terminal output:

Running oc with args [oc get routes --namespace zevxjfovay]
[oc] No resources found in zevxjfovay namespace.
StdOut is : 
```

However you can use ```Err.Contents()``` atm till the upstream issue is not resolved. You can see how the actual output channels to StdErr instead of StdOut though the exit code is ```0```. Then only your string validation will make sense.
```
func (oc *OcRunner) VerifyResourceDeleted(resourceType, resourceName, namespace string) {
	session := CmdRunner(oc.path, "get", resourceType, "--namespace", namespace)
	Eventually(session).Should(gexec.Exit(0))
	output := string(session.Wait().Err.Contents())
	fmt.Println("StdErr is :", output)
	Expect(output).NotTo(ContainSubstring(resourceName))
}

Terminal output:

Running oc with args [oc get routes --namespace mqdtjckwfe]
[oc] No resources found in mqdtjckwfe namespace.
StdErr is : No resources found in mqdtjckwfe namespace.
```

Reply:
@amitkrout it does not output `""` if the resource exist, if the resource exist and not deleted it will list the resources in the content and 	`Expect(output).NotTo(ContainSubstring(resourceName))` would fail

Reply:
> it does not output ""

Yes, it gives the output ```""```` in ```StdOut``` channel because of the upstream issue.

Pasting it again. Please check it.
```
func (oc *OcRunner) VerifyResourceDeleted(resourceType, resourceName, namespace string) {
	session := CmdRunner(oc.path, "get", resourceType, "--namespace", namespace)
	Eventually(session).Should(gexec.Exit(0))
	output := string(session.Wait().Out.Contents())
	fmt.Println("StdOut is :", output)
	Expect(output).NotTo(ContainSubstring(resourceName))
}

Terminal output:

Running oc with args [oc get routes --namespace zevxjfovay]
[oc] No resources found in zevxjfovay namespace.
StdOut is :
```

Check for more info - https://stackoverflow.com/questions/3385201/confused-about-stdin-stdout-and-stderr

Reply:
**Test gets failed, when it found any resource is existed. I see nothing wrong in the test case.**

`odo sub component command tests
/home/adisky/go/src/github.com/openshift/odo/tests/integration/cmd_cmp_sub_test.go:13
  odo component delete should clean owned resources
  /home/adisky/go/src/github.com/openshift/odo/tests/integration/component.go:823
    should delete the component and the owned resources with wait flag [It]
    /home/adisky/go/src/github.com/openshift/odo/tests/integration/component.go:858

    Expected
        <string>: NAME              HOST/PORT                                     PATH   SERVICES      PORT   TERMINATION   WILDCARD
        example-1-zgpbk   example-1-zgpbk-fgqbtmujjt.apps-crc.testing          lfhyf-zgpbk   8080                 None
        example-2-zgpbk   example-2-zgpbk-fgqbtmujjt.apps-crc.testing          lfhyf-zgpbk   8080                 None
        
    not to contain substring
        <string>: example

    /home/adisky/go/src/github.com/openshift/odo/tests/helper/helper_oc.go:491
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 2 Failures:

[Fail] odo component command tests odo component delete should clean owned resources [It] should delete the component and the owned resources with wait flag 
/home/adisky/go/src/github.com/openshift/odo/tests/helper/helper_oc.go:491
`

Reply:
> > it does not output ""
> 
> Yes, it gives the output `""` in `StdOut` channel because of the upstream issue.
> 
> Pasting it again. Please check it.
> 
> ```
> func (oc *OcRunner) VerifyResourceDeleted(resourceType, resourceName, namespace string) {
> 	session := CmdRunner(oc.path, "get", resourceType, "--namespace", namespace)
> 	Eventually(session).Should(gexec.Exit(0))
> 	output := string(session.Wait().Out.Contents())
> 	fmt.Println("StdOut is :", output)
> 	Expect(output).NotTo(ContainSubstring(resourceName))
> }
> 
> Terminal output:
> 
> Running oc with args [oc get routes --namespace zevxjfovay]
> [oc] No resources found in zevxjfovay namespace.
> StdOut is :
> ```
> 
> Check for more info - https://stackoverflow.com/questions/3385201/confused-about-stdin-stdout-and-stderr
It outputs the resource name when it exist.

Reply:
Yup, this is expected because of https://github.com/openshift/odo/blob/master/tests/helper/helper_run.go#L77

If fails it get the output from ```StdErr``` like mentioned in line https://github.com/openshift/odo/blob/master/tests/helper/helper_run.go#L80